ingress/controllers/nginx/README.md

```diff
@@ -149,6 +150,23 @@ By default the controller redirects (301) to HTTPS if there is a TLS Ingress rul
 To disable this behavior use `hsts=false` in the NGINX config map.
 
 
+### Automated Certificate Management with Kube-Lego
+
+[Kube-Lego] automatically requests missing certificates or expired from
+[Let's Encrypt] by monitoring ingress resources and its referenced secrets. To
+enable this for an ingress resource you have to add an annotation:
+
+```
+kubectl annotate ing ingress-demo kubernetes.io/tls-acme="true"
```

Comment:
Does AWS support Ingress? 
Currently there is no policy around how each backend handles a new ingress. Some like the nginx controller write out a single config with all the rules, others like the gce controller (and vshphere i believe) will allocate a vip per ingress. I don't know how we're planning to do it with aws yet (https://github.com/kubernetes/contrib/pull/841).

There should be something at the api layer that specifies if the Ingress "joins" an existing hostname/ip, or needs a new one. We don't have that yet.